cpp/test/sparse/preprocess_coo.cu

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "../test_utils.cuh"
+
+#include <raft/core/device_coo_matrix.hpp>
+#include <raft/core/resource/cuda_stream.hpp>
+#include <raft/core/resources.hpp>
+#include <raft/sparse/matrix/preprocessing.cuh>
+#include <raft/sparse/selection/knn.cuh>
+#include <raft/util/cudart_utils.hpp>
+
+#include <gtest/gtest.h>
+
+#include <iostream>
+#include <limits>
+
+namespace raft {
+namespace sparse {
+
+template <typename Type_f, typename Index_>
+struct SparsePreprocessInputs {
+  int n_rows;
+  int n_cols;
+  std::vector<Index_> rows_h;
+  std::vector<Index_> columns_h;
+  std::vector<Type_f> values_h;
+};
+
+template <typename Type_f, typename Index_>
+class SparsePreprocessCoo
+  : public ::testing::TestWithParam<SparsePreprocessInputs<Type_f, Index_>> {
+ public:
+  SparsePreprocessCoo()
+    : params(::testing::TestWithParam<SparsePreprocessInputs<Type_f, Index_>>::GetParam()),
+      stream(resource::get_cuda_stream(handle)),
+      n_rows(params.n_rows),
+      n_cols(params.n_cols),
+      rows(params.rows_h.size(), stream),
+      columns(params.columns_h.size(), stream),
+      values(params.values_h.size(), stream),
+      result(params.values_h.size(), stream)
+  {
+  }
+
+ protected:
+  void SetUp() override {}
+
+  void Run(bool bm25_on)
+  {
+    int k                                 = 2;
+    std::vector<Type_f> bm25_vals_h       = {1.03581,
```

Comment:
all of these values were added here, because when I tried to add them as params I kept getting segfaults.